src/csv-report/ghazdo-csv-report.ps1

```diff
@@ -0,0 +1,151 @@
+    - BUILD_REPOSITORY_NAME: The name of the Azure DevOps repository.
+    - BUILD_BUILDNUMBER: The build number of the Azure DevOps build.
+#>
+$pass = ${env:MAPPED_ADO_PAT} #$env:MAPPED_ADO_PAT = "TODO-ADVSEC-SCOPED-PAT-HERE"
+$orgUri = ${env:SYSTEM_COLLECTIONURI} #$(System.CollectionUri) #$env:SYSTEM_COLLECTIONURI = "https://dev.azure.com/TODO-YOUR-ORG-HERE/"
+$orgName = $orgUri -replace "^https://dev.azure.com/|/$"
+$project = ${env:SYSTEM_TEAMPROJECT} #$(System.TeamProject) #$env:SYSTEM_TEAMPROJECT = "TODO-YOUR-PROJECT-NAME-HERE"
+$repositoryId = ${env:BUILD_REPOSITORY_ID} #$(Build.Repository.ID) #$env:BUILD_REPOSITORY_ID= "TODO-YOUR-REPO-GUID-HERE"
+$repositoryName = ${env:BUILD_REPOSITORY_NAME} #$(Build.Repository.Name) #$env:BUILD_REPOSITORY_NAME= "TODO-YOUR-REPO-NAME-HERE"
+$build = ${env:BUILD_BUILDNUMBER} # $env:BUILD_BUILDNUMBER= "$(Get-Date -Format "yyyyMMdd").1"
+
+$headers = @{ Authorization = "Basic $([System.Convert]::ToBase64String([System.Text.Encoding]::ASCII.GetBytes(":$pass")))"; }
+
+# Report Configuration
+$allRepos = $true #run for all repos in the org/projects
+$severities = @("critical", "high", "medium", "low")
+$states = @("active", "fixed", "dismissed")
+$alertTypes = @("code", "secret", "dependency")
+$severityDays = @{
+    "critical" = 7
+    "high"     = 30
+    "medium"   = 90
+    "low"      = 180
+}
+
+# get list of projects in the Organization
+$url = "https://dev.azure.com/{0}/_apis/projects?api-version=6.0" -f $orgName
+$projectsResponse = Invoke-WebRequest -Uri $url -Headers $headers -Method Get
+$projects = ($projectsResponse.Content | ConvertFrom-Json).value
+
+# create a hashtable to hold org name, project name and repo id
+$scans = @()
+
+
+if ($allRepos) {
+    foreach ($proj in $projects) {
+        $url = "https://dev.azure.com/{0}/{1}/_apis/git/repositories?api-version=6.0" -f $orgName, $proj.name
+        $reposResponse = Invoke-WebRequest -Uri $url -Headers $headers -Method Get
+        $repos = ($reposResponse.Content | ConvertFrom-Json).value
+        # Add the org name, project name, and repo ID to the hashtable for each repository
+        foreach ($repo in $repos) {
+            $scans += @{
+                OrgName     = $orgName
+                ProjectName = $proj.name
+                RepoName    = $repo.name
+                RepoId      = $repo.id
+            }
+        }
+    }
+}
+else {
+    $scans += @{
+        OrgName     = $orgName
+        ProjectName = $project
+        RepoName    = $repositoryName
+        RepoId      = $repositoryId
+    }
+}
+
+
+
+#loop through repo alert list
+[System.Collections.ArrayList]$alertList = @()
+foreach ($scan in $scans) {
+    $project = $scan.ProjectName
+    $repositoryName = $scan.RepoName
+    $repositoryId = $scan.RepoId
+    $alerts = $null
+    $parsedAlerts = $null
+    $url = "https://advsec.dev.azure.com/{0}/{1}/_apis/AdvancedSecurity/Repositories/{2}/alerts?useDatabaseProvider=true" -f $orgName, $project, $repositoryId
```

Comment:
the useDatabaseProvider flag has been deprecated, and we shouldn't need it at this point.